### PR TITLE
Simplify profiler impl (bubble up Option)

### PR DESCRIPTION
**Pull request by Veetaha**

### CODE REVIEW

**Comment by Veetaha** [comment]
Force-pushet due to some rustup connection problem on CI:

File: crates/ra_prof/src/hprof.rs [hunk 1]
```diff
@@ -58,36 +59,33 @@ pub type Label = &'static str;
 /// ```
 pub fn profile(label: Label) -> Profiler {
     assert!(!label.is_empty());
-    let enabled = PROFILING_ENABLED.load(Ordering::Relaxed)
-        && PROFILE_STACK.with(|stack| stack.borrow_mut().push(label));
-    let label = if enabled { Some(label) } else { None };
-    Profiler { label, detail: None }
+    if PROFILING_ENABLED.load(Ordering::Relaxed) {
+        PROFILE_STACK.with(|stack| stack.borrow_mut().push(label));
```
Comment (matklad) [inline crates/ra_prof/src/hprof.rs:63]: This changes semantics, it’ll return non trivial profiler where the old code would use None label

Comment (Veetaha) [reply crates/ra_prof/src/hprof.rs:63]: What do you mean by non-trivial profiler, it is just a newtype of `Option<Impl>`

Comment (matklad) [reply crates/ra_prof/src/hprof.rs:63]: Reviewing from my iPad, so might be wrong, but it still seems to me that the refactoring changes behavior. && was meaningful.

Comment (Veetaha) [reply crates/ra_prof/src/hprof.rs]: You are right, I mistakenly thought that the boolean value returned from `push` is unused, already fixed it.
Regarding the non-trivilaity, the [`rustc`](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=e1cb2f91b2117de70c951c336ce13dc4) is smart enough not to increase the size of the profiler and use the inner string reference non-nullability to store the discriminant.

**Comment by Veetaha** [comment]
@matklad, this one is easy, do we merge it?

**Comment by lnicola** [comment]
Might need a rebase.

**Pull request merged**